exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceArtifactId.java

```diff
@@ -25,29 +25,34 @@
 
 /**
  * A service artifact identifier. It consists of the runtime id in which the service shall be
- * deployed and the service artifact name. The name of Java artifacts usually contains the three
- * coordinates identifying any Java artifact: groupId, artifactId and version.
+ * deployed, the service artifact name and the artifact version. The name of Java
+ * artifacts usually consists of groupId and artifactId.
  *
  * <p>The extensions of this class must be immutable and hence thread-safe.
  */
 @AutoValue
 public abstract class ServiceArtifactId {
 
   private static final String DELIMITER = ":";
-  private static final int NUM_FIELDS = 2;
+  private static final int NUM_FIELDS = 3;
 
   /**
    * Returns the runtime id in which the service shall be deployed.
    */
   public abstract int getRuntimeId();
 
   /**
-   * Returns the full artifact name of this service (e.g., "com.acme:land-registry:1.2.0").
+   * Returns the full artifact name of this service (e.g., "com.acme:land-registry").
```

Comment:
You can't expect the rest of the code to work by using ':' as delimiter for groupId and artifactId.